src/Raven.Server/Documents/PeriodicBackup/BackupHelper.cs

```diff
@@ -29,5 +34,39 @@ public static bool BackupTypeChanged(PeriodicBackupConfiguration previous, Perio
         {
             return previous.BackupType != current.BackupType;
         }
+
+        public static async Task<long> RunWithRetriesAsync(
+            int maxRetries,
+            Func<Task<long>> action,
+            string infoMessage,
+            string errorMessage,
+            SmugglerResult smugglerResult,
+            Action<IOperationProgress> onProgress = default,
+            OperationCancelToken operationCancelToken = default)
+        {
+            var retries = 0;
+
+            while (true)
+            {
+                try
+                {
+                    operationCancelToken?.Token.ThrowIfCancellationRequested();
```

Comment:
Let's throw only after we tried to execute the command at least once.

Reply:
Done.